cmd/thanos/query.go

```diff
@@ -317,20 +321,30 @@ func runQuery(
 			maxConcurrentSelects,
 			queryTimeout,
 		)
-		engine = promql.NewEngine(
-			promql.EngineOpts{
-				Logger: logger,
-				Reg:    reg,
-				// TODO(bwplotka): Expose this as a flag: https://github.com/thanos-io/thanos/issues/703.
-				MaxSamples: math.MaxInt32,
-				Timeout:    queryTimeout,
-				NoStepSubqueryIntervalFn: func(rangeMillis int64) int64 {
-					return defaultEvaluationInterval.Milliseconds()
+		subqueryIntervalFunc = func(int64) int64 {
+			return defaultEvaluationInterval.Milliseconds()
+		}
+		totalEngines int
+		newEngine    = func(ld time.Duration) *promql.Engine {
```

Comment:
I am a bit confused, do we need two functions? What about `engineFunc`? I guess you wanted to have large number of args in function, but it might be cleaner to embed `engineFunc` here, not embedding this function in `engineFunc`. WDYT?

Reply:
You're absolutely right, having two functions is confusing. Got some coffee, merged engineFunc and newEngine into single engineFactory. IMO things got a lot more unerstandable. I hope this isn't just coffee speaking.